js/initcontext_test.go

```diff
@@ -241,70 +241,88 @@ func TestInitContextRequire(t *testing.T) {
 	})
 }
 
-func TestInitContextOpen(t *testing.T) {
+func createFile(t *testing.T, file string, content string, binary bool) (*BundleInstance, error) {
 	fs := afero.NewMemMapFs()
 	assert.NoError(t, fs.MkdirAll("/path/to", 0755))
-	assert.NoError(t, afero.WriteFile(fs, "/path/to/file.txt", []byte("hi!"), 0644))
+	assert.NoError(t, afero.WriteFile(fs, "/path/to/"+file, []byte(content), 0644))
 
-	testdata := map[string]string{
-		"Absolute": "/path/to/file.txt",
-		"Relative": "./file.txt",
+	afero.WriteFile(fs, "/path/to/"+file, []byte(content), 0644)
+	binaryArg := ""
+	if binary {
+		binaryArg = ",\"b\""
 	}
-	for name, loadPath := range testdata {
-		t.Run(name, func(t *testing.T) {
-			b, err := NewBundle(&lib.SourceData{
-				Filename: "/path/to/script.js",
-				Data: []byte(fmt.Sprintf(`
-				export let data = open("%s");
+
+	b, err := NewBundle(&lib.SourceData{
+		Filename: "/path/to/script.js",
+		Data: []byte(fmt.Sprintf(`
+				export let data = open("/path/to/%s"%s);
 				export default function() {}
```

Comment:
You might want to do something with the data, for example - check if the `data.length` matches the expected size and throw an exception if it doesn't. For strings that should be the number of characters (UTF-16 code points, important since Go uses utf-8), while for binary files it should just be the byte length.

Reply:
Hi, I added length checking.
But, as far as I know, there is no support of UTF-16 and open() reads the file as UTF-8. 
So I commented line 288 with UTF-16 testing.
We can add UTF-16 support and uncomment that line.

Reply:
Yes, Go strings are UTF-8 encoded (though it's possible to have invalid UTF-8 sequences in a `string`), but strings in JavaScript runtimes (including the [goja runtime](https://github.com/dop251/goja/) we use in k6) are generally UTF-16 or UCS-2 encoded. Though these details probably don't matter much in this case, conversions from UTF-8 to UTF-16 and back should be transparently handled by goja without issues.

What I meant in my previous comment is that when you have `var str = open("/some/text-file");`, `str` is supposed to be a JavaScript string with a `length` equal to the length of the text in the file. If you open a binary file that way (`var strBin = open("/some/bin-file");`), `strBin` would still be a string variable in JavaScript, but it's `lenght` likely won't be the number of bytes in the file due to conversion of invalid bytes to UTF-16. But if you open the file as a binary ([docs](https://docs.k6.io/docs/open-filepath-mode)) with `var bin = open("/some/bin-file", "b")`, `bin` should be a JavaScript array with a length exactly equal to the number of bytes in the binary file.